src/hardware/mod.rs

```diff
@@ -44,33 +49,55 @@ pub trait Reset {
 pub(crate) mod fake {
     use core::convert::TryInto;
     use core::time::Duration;
+    use std::collections::HashMap;
 
     /// A fake `Identity` that returns fixed values.
     pub struct Identity {
         firmware_version: Vec<u8>,
+        vendor_firmware_versions: HashMap<u8, Vec<u8>>,
         unique_id: Vec<u8>,
     }
 
     impl Identity {
         /// Creates a new `fake::Identity`.
-        pub fn new(firmware_version: &[u8], unique_id: &[u8]) -> Self {
-            let mut firmware_version = firmware_version.to_vec();
-            while firmware_version.len() < 32 {
-                firmware_version.push(0);
+        pub fn new(
+            firmware_version: &[u8],
+            vendor_firmware_versions: HashMap<u8, &[u8]>,
+            unique_id: &[u8],
+        ) -> Self {
+            let mut vendor_fw_versions_vec = HashMap::new();
+            for (key, val) in vendor_firmware_versions.iter() {
+                vendor_fw_versions_vec.insert(*key, Self::u8_to_sized_vec(val));
             }
-            firmware_version.truncate(32);
 
             Self {
-                firmware_version,
+                firmware_version: Self::u8_to_sized_vec(firmware_version),
+                vendor_firmware_versions: vendor_fw_versions_vec,
                 unique_id: unique_id.to_vec(),
             }
         }
+
+        fn u8_to_sized_vec(data: &[u8]) -> Vec<u8> {
+            let mut vec = data.to_vec();
+            while vec.len() < 32 {
+                vec.push(0);
+            }
+            vec.truncate(32);
+
+            vec
+        }
     }
 
     impl super::Identity for Identity {
         fn firmware_version(&self) -> &[u8; 32] {
             self.firmware_version[..32].try_into().unwrap()
         }
+        fn vendor_firmware_version(&self, slot: u8) -> Option<&[u8; 32]> {
+            match self.vendor_firmware_versions.get(&slot) {
```

Comment:
How about:

```rust
self.vendor_firmware_versions.get(&slot).map(|data| data[..32].try_into().unwrap())
```

Although perhaps it makes more sense to make the fields of this struct use `[u8; 32]` instead of `Vec<u8>`...